annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,321 @@
+import re
+from unittest.mock import Mock, call, patch
+
+import pytest
+from fastapi import HTTPException
+
+from annotation.errors import FieldConstraintError
+from annotation.jobs.services import ValidationSchema
+from annotation.models import File, ManualAnnotationTask
+from annotation.tasks.services import (
+    check_cross_annotating_pages,
+    validate_files_info,
+    validate_task_info,
+    validate_user_actions,
+    validate_users_info,
+)
+
+
+@pytest.mark.parametrize(
+    ("validation_type", "is_validation"),
+    (
+        (ValidationSchema.validation_only, True),
+        ("other_type", False),
+        ("other_type", True),
+    ),
+)
+def test_validate_task_info(
+    validation_type: ValidationSchema,
+    is_validation: bool,
+):
+    with patch("sqlalchemy.orm.Session", spec=True) as mock_session, patch(
+        "annotation.tasks.services.validate_users_info"
+    ) as mock_validate_users_info, patch(
+        "annotation.tasks.services.validate_files_info"
+    ) as mock_validate_files_info:
+
+        db_session = mock_session()
```

Comment:
Could you elaborate why do that? why not simply use mock_session everywhere?

Reply:
Since I've mocked these two functions, there's no need to use db_session here, as those functions won't be interacting with it. Therefore, I'll remove that part and use None instead.

Reply:
Didnt get the answer and the code is still the same

Reply:
I've made the changes as it seemed more appropriate to write two mock objects for the functions being called. This approach felt more readable and understandable to me, especially since db_session is used for database connection, and I didn't want to add extra functionality. I was aiming to adhere to the single responsibility principle.